assets/js/base/context/hooks/use-customer-data.ts

```diff
@@ -86,12 +89,40 @@ export const useCustomerData = (): {
 		billingAddress: CartResponseBillingAddress;
 	} = useStoreCart();
 
+	// We need to store a ref to the shipping and billing addresses in order to re-populate the local state
+	// here with the hydrated values once they change. The refs keep track of the changing values of the addresses
+	// as they are hydrated.
+	const billingDataRef = useRef( initialBillingAddress );
+	const shippingAddressRef = useRef( initialShippingAddress );
+	billingDataRef.current = initialBillingAddress;
+	shippingAddressRef.current = initialShippingAddress;
```

Comment:
Why are doing this here if we have `hasCustomerDataSynced` and the condition below?

Reply:
It's because `initialBillingAddress` and `initialShippingAddress` will update once the cart store has been hydrated. So we are storing a ref to it to reflect the change

Reply:
Shouldn't the initial value passed to `useRef` serve as that? I know that this line will execute each time the hook runs so are you updating the ref each time?
```
billingDataRef.current = initialBillingAddress;
```